### PR TITLE
Update data-jpa sample with NamedEntityGraph

### PR DESCRIPTION
**Pull request by christophstrobl**
Relates to: spring-projects-experimental/spring-native#1728

### CODE REVIEW

File: data-jpa/src/main/java/com/example/data/jpa/BookRepository.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+package com.example.data.jpa;
+
+import com.example.data.jpa.model.Book;
+import org.springframework.data.jpa.repository.EntityGraph;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.ListCrudRepository;
+
+public interface BookRepository extends ListCrudRepository<Book, Long> {
+
+	// @EntityGraph(value = "Book.authors")
```
Comment (mhalbritter) [inline data-jpa/src/main/java/com/example/data/jpa/BookRepository.java:25]: Is this commented out on purpose?

File: data-jpa/src/main/java/com/example/data/jpa/CLR.java [hunk 2]
```diff
@@ -61,6 +66,18 @@ private void findById(List<Author> authors) {
...
 	}
 
+	private void entityGraph() {
+
+		Book book = new Book(null, "Spring in Action");
+		Author author = new Author(null, "Craig Walls", Collections.emptySet());
+		book.getAuthors().add(author);
+		bookRepository.save(book);
+
+		// Book loaded = bookRepository.findByTitleWithNamedGraph("Spring in Action");
```
Comment (mhalbritter) [inline data-jpa/src/main/java/com/example/data/jpa/CLR.java:76]: Is this commented out on purpose?

File: data-jpa/src/main/java/com/example/data/jpa/model/Book.java [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
+import jakarta.persistence.CascadeType;
 import jakarta.persistence.Entity;
 import jakarta.persistence.GeneratedValue;
 import jakarta.persistence.Id;
+import jakarta.persistence.NamedAttributeNode;
+import jakarta.persistence.NamedEntityGraph;
+import jakarta.persistence.OneToMany;
 
 @Entity
+// @NamedEntityGraph(name = "Book.authors", attributeNodes =
```
Comment (mhalbritter) [inline data-jpa/src/main/java/com/example/data/jpa/model/Book.java:16]: Is this commented out on purpose?

File: data-jpa/src/main/java/com/example/data/jpa/CLR.java (hunk 2 above)
Comment (christophstrobl) [reply data-jpa/src/main/java/com/example/data/jpa/CLR.java:76]: yes - see comment below. I can move the entity graph feature to a new sample, but cannot test it within the same configuration.

**Comment by christophstrobl** [comment]
Added a plain hibernate sample that checks `@NamedEntityGraphs` and updated the `data-jpa` one to use dynamic fetch graphs.

File: hibernate/build.gradle [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	id 'org.springframework.aot.smoke-test'
+	id 'org.graalvm.buildtools.native'
+}
+
+dependencies {
+	implementation(platform(org.springframework.boot.gradle.plugin.SpringBootPlugin.BOM_COORDINATES))
+	implementation("org.springframework.boot:spring-boot-starter-jdbc")
+	implementation("org.springframework:spring-orm")
+	implementation("jakarta.persistence:jakarta.persistence-api")
+	implementation("org.hibernate:hibernate-core:6.1.2.Final")
```
Comment (mhalbritter) [inline hibernate/build.gradle:13]: Isn't that version managed by Boot?

File: hibernate/src/test/java/com/example/spring/orm/HibernateTests.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
+package com.example.spring.orm;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class HibernateTests {
+
+	@Test
+	void contextLoads() {
```
Comment (mhalbritter) [inline hibernate/src/test/java/com/example/spring/orm/HibernateTests.java:10]: Can be removed. Every test which is in `src/main/test` now executes on the JVM and in a native image.

File: settings.gradle [hunk 6]
```diff
@@ -71,6 +71,7 @@ include "flyway"
 include "freemarker-webflux"
 include "freemarker-webmvc"
 include "hateoas"
+include "hibernate"
```
Comment (mhalbritter) [inline settings.gradle:74]: Should be added to the CI file, too.

File: hibernate/src/test/java/com/example/spring/orm/HibernateTests.java (hunk 5 above)
Comment (christophstrobl) [reply hibernate/src/test/java/com/example/spring/orm/HibernateTests.java]: removed

File: settings.gradle (hunk 6 above)
Comment (christophstrobl) [reply settings.gradle:74]: done

File: hibernate/build.gradle (hunk 4 above)
Comment (mhalbritter) [reply hibernate/build.gradle:13]: Boot manages `org.hibernate.orm:hibernate-core`. The group ID is different. Are these two different JARs?

Comment (christophstrobl) [reply hibernate/build.gradle:13]: doh my bad - should have looked at the `3.0.0-SNAPSHOT` dependencies, not current 😅 - feel free to alter at your convenience

**Comment by mhalbritter** [comment]
Thanks!

**Pull request merged**